### ISSUE TITLE
ActivityTagsCollection "The collection already contains item with same key 'xxxx''"

### ISSUE THREAD

**Issue by pfab-io** [rebus-org/Rebus.OpenTelemetry#17]
When an activity tag already exists, there's an System.InvalidOperationException: "The collection already contains item with same key 'xxxx''"

<img width="1214" alt="image" src="https://github.com/rebus-org/Rebus.OpenTelemetry/assets/825690/4641aa81-6fd9-4d96-b192-2fb250396392">

checking source code it seems that activity tags are added without checking if they are already present

<img width="651" alt="image" src="https://github.com/rebus-org/Rebus.OpenTelemetry/assets/825690/57e8f732-019c-4a94-ac6e-d27428f5422c">

maybe i'm using OpenTelemetry in the wrong way, but this is blocking because the exception is stopping the main thread of the message

in my case this happens when all the retries of the messages fail and Rebus is preparing to send an IFailed<>

basically since i've added Rebus.OpenTelemetry my app has stopped to receive IFailed<>

**Comment by pfab-io** [comment]
hello, any news to have a new stable  or alpha nuget package with this fix?

latest version is still 1.0.0 (8 monthos ago) and we are blocked to send rebus opentelemetry data to app insights

**Comment by zlepper** [comment]
@mookid8000 Can you make a release? :)

**Comment by mookid8000** [comment]
Yes, sorry! I Didn't realize that there was unreleased stuff in here 😅 

It's out as Rebus.OpenTelemetry 1.0.1 on NuGet.org now!

THanks for waking me up from my slumber 🙂
